tests/stress/km/stress_tests_km.cpp

```diff
@@ -210,6 +197,44 @@ _restart_extension(const std::string& extension_name, uint32_t timeout)
     return status;
 }
 
+static std::thread
+_start_extension_restart_thread(const std::string& extension_name, uint32_t restart_delay, uint32_t thread_lifetime)
+{
+    return std::thread(
+        [&](uint32_t local_restart_delay, uint32_t local_thread_lifetime) {
+            // Delay the start of this thread for a bit to allow the ebpf programs to attach successfully. There's a
+            // window where if the extension is unloading/unloaded, an incoming attach might fail.
+            std::this_thread::sleep_for(std::chrono::seconds(3));
```

Comment:
This implies the test is non-deterministic and if the CPU is pegged then the test can fail, is that correct?  If so, there's probably a better solution in the future than introducing an arbitrary delay.

Reply:
No, the test _itself_ wouldn't fail. It would just carry on with the start of the 'extension restart' thread delayed even further.  Given that the minimum test runtime is 1 minute, IMO it's more than enough for the restart thread to start running eventually.   In any case, system load affecting thread scheduling would affect the test threads as well, so their delays would balance out.

FWIW, I _did_ think of synchronizing this thread with the test threads but thought it was overkill given the 'test only' nature of this code.  But I can certainly file an issue to revisit that synchronization if it is seen to cause issues during actual testing.